### PR TITLE
K8s ClusterRoles for: TFJobs, PyTorchJobs, Notebooks, PodDefaults, Profiles

### PR DESCRIPTION
**Pull request by krishnadurai**
**Which issue is resolved by this Pull Request:**
Partially Addresses: kubeflow/kubeflow#3938

**Description of your changes:**
Read Issue description: kubeflow/kubeflow#3938
Related Issues: kubeflow/kubeflow#4167 #374 
This PR enables a user created through Profiles or Kfam to access the following Custom Resouces within their namespace:

TFJobs, PyTorchJobs, Notebooks, PodDefaults, Profiles

The above ClusterRoles are aggregated to Kubeflow Roles mentioned in #374

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate`
    3. `make test`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/393)

### CODE REVIEW

File: admission-webhook/webhook/base/cluster-role.yaml [hunk 1]
```diff
@@ -15,3 +15,59 @@ rules:
   - create
   - patch
   - delete
+
```
Comment (kunmingg) [inline admission-webhook/webhook/base/cluster-role.yaml]: Editors shouldn't have permission to change poddefaults.
Those are reserved for cluster admin who setup like cloud credentials.

File: profiles/base/cluster-role.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+---
```
Comment (kunmingg) [inline profiles/base/cluster-role.yaml]: Same here
Profiles are not namespace-scoped, we don't need to aggregate them.

**Comment by krishnadurai** [comment]
> And shall we add role for katib?
Sure, @kunmingg I'm looking to have separate PRs for different components because the reviewers might be different. This is necessary because some components like Katib are looking to add new CRs and are not already a part of our deployments or manifests.

File: admission-webhook/webhook/base/cluster-role.yaml (hunk 1 above)
Comment (krishnadurai) [reply admission-webhook/webhook/base/cluster-role.yaml]: Thanks for providing this context @kunmingg, I'll address it.

File: profiles/base/cluster-role.yaml (hunk 2 above)
Comment (krishnadurai) [reply profiles/base/cluster-role.yaml]: I'll make this change.

**Comment by krishnadurai** [comment]
@kunmingg I've addressed review changes.

**Comment by kunmingg** [comment]
/lgtm
/approve
Thanks!

**Comment by krishnadurai** [comment]
@richardsliu @johnugeorge could you please take a look to see if the roles are appropriate?

File: tf-training/tf-job-operator/base/cluster-role.yaml [hunk 3]
```diff
@@ -104,3 +104,63 @@ rules:
...
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kubeflow-tfjobs-view
+  labels:
+    rbac.authorization.kubeflow.org/aggregate-to-kubeflow-view: "true"
+rules:
+- apiGroups:
+  - tensorflow.org
```
Comment (johnugeorge) [inline tf-training/tf-job-operator/base/cluster-role.yaml]: I see this even in https://github.com/kubeflow/manifests/blob/master/tf-training/tf-job-operator/base/cluster-role.yaml#L64 

Not sure why this is needed. We can revisit later.  
@richardsliu

**Comment by krishnadurai** [comment]
@richardsliu can you PTAL?

**Comment by krishnadurai** [comment]
Can someone add an LGTM please?

**Pull request merged**